lib/protocol/protocol.go

```diff
@@ -662,12 +664,13 @@ func (c *rawConnection) send(msg message, done chan struct{}) bool {
 	select {
 	case c.outbox <- asyncMessage{msg, done}:
 		return true
+	case <-c.sendStop:
```

Comment:
Slightly confusing naming - we're not sending a stop, we're stopping the sender? `preventSends`? I dunno. Maybe leave as is.